crates/kernels/vulkan-and-cpu/src/tests/matchers.rs

```diff
@@ -0,0 +1,44 @@
+//! A custom matcher for matching coordinates.
+
+#![cfg(test)]
+
+use googletest::prelude::*;
+
+#[derive(MatcherBase)]
+/// A custom matcher for coordinates. Really only needed to match NaNs.
+pub struct CoordMatcher {
+    expected: glam::Vec2,
+}
+
+impl Matcher<glam::Vec2> for CoordMatcher {
+    fn matches(&self, actual: glam::Vec2) -> googletest::matcher::MatcherResult {
+        let x_nan = actual.x.is_nan() && self.expected.x.is_nan();
```

Comment:
IEEE754 once again bamboozling us.

This doesn't handle a +NaN/-NaN, but do we care? If we did you just have to compare the signs before these comparisions